pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This file defines the scheduling framework plugin interfaces.
+
+package v1alpha1
+
+import (
+	"k8s.io/api/core/v1"
+	internalcache "k8s.io/kubernetes/pkg/scheduler/internal/cache"
+)
+
+// Plugin is the parent type for all the scheduling framework plugins.
+type Plugin interface {
+	Name() string
+}
+
+// ReservePlugin is an interface for Reserve plugins. These plugins are called
+// at the reservation point, AKA "assume". These are meant to updated the state
+// of the plugin. They do not return any value (other than error).
+type ReservePlugin interface {
+	Plugin
+	// Reserve is called by the scheduling framework when the scheduler cache is
+	// updated.
+	Reserve(pc *PluginContext, p *v1.Pod, nodeName string) error
+}
+
+// PrebindPlugin is an interface that must be implemented by "prebind" plugins.
+// These plugins are called before a pod being scheduled
+type PrebindPlugin interface {
+	Plugin
+	// Prebind is called before binding a pod. All prebind plugins must return
+	// err = nil or the pod will be rejected and won't be sent for binding.
+	// If the error is not of type "unschedulable", it is considered as an internal
+	// error.
+	Prebind(pc *PluginContext, p *v1.Pod, nodeName string) error
+}
+
+type unschedulable struct{ error }
+
+// Unschedulable makes an "unschedulable" error out of the given "err". This is
```

Comment:
Here is a suggested rewording.

Unschedulable returns an error indicating that a pod cannot be scheduled. It can be used by plugins to report when a pod fails to meet some constraint.